openwrt-openwisp-monitoring/files/monitoring.agent

```diff
@@ -26,76 +29,184 @@ if [ "$VERSION" -eq "1" ]; then
   exit 0
 fi
 
-VERIFY_SSL=${VERIFY_SSL:-0}
 INTERVAL=${INTERVAL:-300}
-MONITORED_INTERFACES=${MONITORED_INTERFACES:-*}
 verbose_mode=${verbose_mode:-0}
+TMP_DIR="/tmp/openwisp/monitoring"
 
-if [ -z "$BASE_URL" ]; then
-  logger -s "missing required --url option" \
-         -t openwisp_monitoring \
-         -p daemon.err
-  exit 1
-fi
-
-if [ -z "$UUID" ]; then
-  logger -s "missing required --uuid option" \
-          -t openwisp_monitoring \
-          -p daemon.err
-  exit 1
-fi
-
-if [ -z "$KEY" ]; then
-  logger -s "missing required --key option" \
-         -t openwisp_monitoring \
-         -p daemon.err
-  exit 1
-fi
-
-# Remove double quotes from interfaces
-MONITORED_INTERFACES=$(echo "$MONITORED_INTERFACES" | tr -d '"')
-
-URL="$BASE_URL/api/v1/monitoring/device/$UUID/?key=$KEY"
+check_available_memory(){
+  total=$(ubus call system info | jsonfilter -e '@.memory.total')
+  available=$(ubus call system info | jsonfilter -e '@.memory.available')
+  required=$(echo - |awk -v percent="$REQUIRED_PERCENT" -v total="$total" '{printf("%.f",percent*total)}')
```

Comment:
Is there a way to improve this too? I tried searching for one but end in futile.

Reply:
at the moment I don't have ideas on how to further improve this.